### PR TITLE
Fix/pi 2658/add set query args helper

### PR DESCRIPTION
**Pull request by rblaa**
add url query args update support

### CODE REVIEW

**Review by bdachev0710 — APPROVED** [review]

File: src/utils/url_params.js [hunk 1]
```diff
@@ -1,19 +1,54 @@
 /**
- * parseQueryArgs - parses the query args from a given url
- * @url {string} - string of the url to parse the query args from
- * @separatorString {string} - string to separate the query args by
- * @paramChar {string} - string to split the url from the query args
- * @decodeArgs {boolean} - boolean to determine decoding the query args
+ * Parses the query args from a given url. Both #..&.. hash args as well as ?..&.. query args are supported.
+ * @param url {string} - string of the url to parse the query args from
```
Comment (7thsky) [inline src/utils/url_params.js]: js-docs defines function parameter tag as `@param {<type>} <name>` - https://jsdoc.app/tags-param

File: src/utils/url_params.js [hunk 2]
```diff
@@ -1,19 +1,54 @@
...
+    const existingArgs = parseQueryArgs(url, separatorString, paramChar);
+    if (!params) params = {};
+    const updatedArgs = {...existingArgs, ...params};
+    return setQueryArgs(url, updatedArgs, separatorString, paramChar);
+}
+
+/**
+ * Sets the url's query args to the specified values. Both #..&.. hash args as well as ?..&.. query args are supported.
+ * @param url - the url to update
+ * @param params {Object} - key/value parameter map to include * @param separatorString
```
Comment (7thsky) [inline src/utils/url_params.js]: merged two lines ( "@param param {Object}" and "* @param separatorString )

**Review by 7thsky — COMMENTED** [review]
js-docs improvements

File: src/utils/url_params.js (hunk 1 above)
Comment (rblaa) [reply src/utils/url_params.js]: ah right. thx, will fix.

File: src/utils/url_params.js (hunk 2 above)
Comment (rblaa) [reply src/utils/url_params.js]: good catch

**Pull request merged**